commcare_connect/opportunity/api/serializers.py

```diff
@@ -34,12 +55,26 @@ class Meta:
             "budget_per_visit",
             "total_budget",
             "claim",
+            "learn_progress",
         ]
 
     def get_claim(self, obj):
+        opp_access = self._get_opp_access(obj)
+        claim = OpportunityClaim.objects.filter(opportunity_access=opp_access)
+        if claim.exists():
+            return OpportunityClaimSerializer(claim.first()).data
+        return None
+
+    def get_learn_progress(self, obj):
+        opp_access = self._get_opp_access(obj)
+        total_modules = LearnModule.objects.filter(app=opp_access.opportunity.learn_app)
+        completed_modules = CompletedModule.objects.filter(opportunity=opp_access.opportunity)
+        return {"total_modules": total_modules.count(), "completed_modules": completed_modules.count()}
+
+    def _get_opp_access(self, obj):
```

Comment:
This is getting called twice now so it should be cached / memoized to prevent multiple DB queries.

Reply:
Can i use the `quickcache` decorator to cache this method?